callable/src/main/java/io/grpc/callable/Callable.java

```diff
@@ -0,0 +1,423 @@
+   * </pre>
+   */
+  public static <RequestT, ResponseT> Callable<RequestT, ResponseT>
+      create(MethodDescriptor<RequestT, ResponseT> descriptor) {
+    return create(CallableDescriptor.create(descriptor));
+  }
+
+  /**
+   * Returns a callable which executes the method described by a {@link CallableDescriptor}.
+   */
+  public static <RequestT, ResponseT> Callable<RequestT, ResponseT>
+      create(final CallableDescriptor<RequestT, ResponseT> descriptor) {
+    return new Callable<RequestT, ResponseT>() {
+      @Override public ClientCall<RequestT, ResponseT> newCall(Channel channel) {
+        if (channel == null) {
+          throw new IllegalStateException(String.format(
+              "unbound callable for method '%s' requires a channel for execution",
+              descriptor.getMethodDescriptor().getFullMethodName()));
+        }
+        return channel.newCall(descriptor.getMethodDescriptor(), CallOptions.DEFAULT);
+      }
+
+      @Override public CallableDescriptor<RequestT, ResponseT> getDescriptor() {
+        return descriptor;
+      }
+
+      @Override public String toString() {
+        return descriptor.getMethodDescriptor().getFullMethodName();
+      }
+    };
+  }
+
+  /**
+   * Returns a callable which executes the given function asynchronously on each provided
+   * input. The supplied executor is used for creating tasks for each input. Example:
+   *
+   * <pre>
+   *  Callable.Transformer&lt;RequestT, ResponseT> transformer = ...;
+   *  Response response = Callable.create(transformer, executor).call(channel, request);
+   * </pre>
+   */
+  public static <RequestT, ResponseT> Callable<RequestT, ResponseT>
+      create(Transformer<? super RequestT, ? extends ResponseT> transformer, Executor executor) {
+    return new TransformingCallable<RequestT, ResponseT>(transformer, executor);
+  }
+
+
+  /**
+   * Returns a callable which executes the given function immediately on each provided input.
+   * Similar as {@link #create(Transformer, Executor)} but does not operate asynchronously and does
+   * not require an executor.
+   *
+   * <p>Note that the callable returned by this method does not respect flow control. Some
+   * operations applied to it may deadlock because of this. However, it is safe to use this
+   * callable in the context of a {@link #followedBy(Callable)} operation, which is the major
+   * use cases for transformers. But if you use a transformer to simulate a real rpc
+   * you should use {@link #create(Transformer, Executor)} instead.
+   */
+  public static <RequestT, ResponseT> Callable<RequestT, ResponseT>
+      create(Transformer<? super RequestT, ? extends ResponseT> transformer) {
+    return new TransformingCallable<RequestT, ResponseT>(transformer, null);
+  }
+
+  /**
+   * Interface for a transformer. It can throw a {@link StatusException} to indicate an error.
+   */
+  public interface Transformer<RequestT, ResponseT> {
```

Comment:
If I understand right, the scenario you are describing would be achieved by

    rpc1.followedBy(transformer).followedBy(rpc2)

So the transformer is not augmenting a call, but actually representing one (in memory). That turned out to be useful also for testing, and I can imagine it's also for other similar scenarios.

What you are describing is an alternative, but it seems less flexible as it always need a real rpc.